controllers/kyma_controller.go

```diff
@@ -148,16 +155,12 @@ func (r *KymaReconciler) getSyncedContext(ctx context.Context, kyma *v1beta2.Kym
 	ctxWithSync, err := remote.InitializeSyncContext(ctx, kyma,
 		r.RemoteSyncNamespace, remoteClient, r.RemoteClientCache)
 	if err != nil {
-		if !kyma.DeletionTimestamp.IsZero() && errors.Is(err, remote.ErrAccessSecretNotFound) {
-			logger.Info("access secret not found for kyma, assuming already deleted cluster")
-			return nil, r.removeAllFinalizersAndUpdateKyma(ctx, kyma)
-		}
 		if util.IsConnectionRefused(err) {
 			logger.Info("connection refused, assuming connection is invalid and resetting cache-entry for kyma")
 			r.RemoteClientCache.Del(client.ObjectKeyFromObject(kyma))
 		}
 		r.enqueueWarningEvent(kyma, syncContextError, err)
-		return nil, err
+		return ctx, err
```

Comment:
This doesn't follow the golang convention: In case of on non-nil error the other returned values should be considered unreliable by the caller. As such it should be documented. But I'd rather avoid it and just return `nil. err` here.

Reply:
Yes, it's intended, if an error happens, the function should **return back original context**, otherwise, follow-up process will get nil pointer when it uses context.

I updated the function comment to emphasise this in case later we accidentally change it.